### PR TITLE
Tweak: Dynamic content support to page settings import [ED-10263]

### PR DESCRIPTION
**Pull request by ranshoshani34**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Added support for dynamic controls when importing page settings.

## Description
An explanation of what is done in this PR

* Expanded the already existing parsing process of dynamic content in elements after kit import to parse page settings as well, using the control instance that represents the setting.

## Test instructions
This PR can be tested by following these steps:

* Use import feature to test.

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/base/document.php [hunk 1]
```diff
@@ -1207,6 +1207,32 @@ public static function on_import_update_dynamic_content( array $config, array $d
...
+   *
+   * @return array
+   */
+	public function on_import_update_dynamic_settings( array $settings, array $config ): array {
+		$controls = $this->get_controls();
+		$controls_manager = Plugin::$instance->controls_manager;
+
+		foreach ( $settings as $key => $value ) {
+			if ( isset( $controls[ $key ] ) ) {
+				$control = $controls[ $key ];
```
Comment (ManorHazazE) [inline core/base/document.php:1224]: Maybe? More readable.
```suggestion
			if ( ! isset( $controls[ $key ] ) ) {
			continue;
			}
			
			$control = $controls[ $key ];
			...
```

File: core/base/document.php [hunk 2]
```diff
@@ -1207,6 +1207,32 @@ public static function on_import_update_dynamic_content( array $config, array $d
...
+				$control_instance = $controls_manager->get_control( $control['type'] );
+
+				if ( $control_instance ) {
```
Comment (ManorHazazE) [inline core/base/document.php:1227]: Same as above.

**Review by ghost — CHANGES_REQUESTED** [review]
Tests?

File: app/modules/import-export/processes/import.php [hunk 3]
```diff
@@ -52,11 +53,11 @@ class Import {
 	private $adapters;
 
 	/**
-	 * Document's elements that imported during the process.
+	 * Document's content (elements and settings) that imported during the process.
```
Comment (ManorHazazE) [inline app/modules/import-export/processes/import.php:56]:
```suggestion
	 * Document's data (elements and settings) that imported during the process.
```

File: tests/phpunit/elementor/app/import-export/processes/test-import.php [hunk 4]
```diff
@@ -67,7 +68,16 @@ public function test_run__import_all() {
...
 		$this->assertEquals( [ 'Elementor', 'Elementor Pro' ], $result['plugins']);
+
 		$this->assertTrue( $result['site-settings'] );
+
+		// Assert that site settings imported without changes.
+		$kit_id = Plugin::$instance->kits_manager->get_active_id();
+		$kit = Plugin::$instance->documents->get( $kit_id, false );
+		$site_settings = $kit->get_settings();
+		$expected_settings = $this->get_mock_site_settings();
+		$this->assert_array_included_in_array( $expected_settings, $site_settings );
```
Comment (ManorHazazE) [inline tests/phpunit/elementor/app/import-export/processes/test-import.php:79]: Maybe put it all in helper function?

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**